runtime/convertValues.go

```diff
@@ -361,10 +371,23 @@ func exportCompositeValue(
 		for i, field := range fieldNames {
 			fieldName := field.Identifier
 
-			fieldValue := v.GetField(inter, locationRange, fieldName)
-			if fieldValue == nil && v.ComputedFields != nil {
-				if computedField, ok := v.ComputedFields[fieldName]; ok {
-					fieldValue = computedField(inter, locationRange)
+			var fieldValue interpreter.Value
+
+			switch v := v.(type) {
+			case *interpreter.SimpleCompositeValue:
+				fieldValue = v.Fields[fieldName]
+				computeField := v.ComputeField
+
+				if fieldValue == nil && computeField != nil {
+					fieldValue = computeField(fieldName, inter, locationRange)
+				}
+
+			case *interpreter.CompositeValue:
+				fieldValue = v.GetField(inter, locationRange, fieldName)
+				if fieldValue == nil && v.ComputedFields != nil {
+					if computedField, ok := v.ComputedFields[fieldName]; ok {
```

Comment:
What's going on here with this change? Seems like more code is duplicated now?

Reply:
No, this is the result of deduplicating `exportSimpleCompositeValue` below, which was identical to this function, `exportCompositeValue`, except for the logic to get the value's fields.